scipy/optimize/_zeros_py.py

```diff
@@ -1392,3 +1393,344 @@ def toms748(f, a, b, args=(), k=1,
                           maxiter=maxiter, disp=disp)
     x, function_calls, iterations, flag = result
     return _results_select(full_output, (x, function_calls, iterations, flag))
+
+
+def _chandrupatla(func, a, b, *, args=(), xatol=_xtol, xrtol=_rtol,
+                  fatol=None, frtol=0, maxiter=_iter, callback=None):
+    """Find the root of an elementwise function using Chandrupatla's algorithm.
```

Comment:
the doc could say why use this method instead of another. e.g. speed, memory, number of iterations, some other feature, etc.

Can be left for later when this get to the public namespace. This information could get into a table.

Reply:
This is currently the only bracketing root-finder that works elementwise. Like `ridder`, it is supposed to be simpler than the Brent methods but with similar performance, but I think you have to try all of them to know which is more efficient for a given problem. `bisect` is probably only better when the function is not smooth (so there is no point in doing interpolation).

`newton` (with its secant and Halley methods, depending on whether derivatives are known) is vectorized, but in my experience is a bit finicky about shapes. 

There is a table in the docs [here](https://docs.scipy.org/doc/scipy/reference/optimize.html#scalar-functions). I started a PR that would move it into the docs of `root_scalar`, but stopped because the table really needs a complete overhaul. I agree that it's a good thing to do, but if we stopped to fix everything we see...